internal/provider/resource_gitlab_user_gpgkey.go

```diff
@@ -0,0 +1,187 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+var _ = registerResource("gitlab_user_gpgkey", func() *schema.Resource {
+	return &schema.Resource{
+		Description: `The ` + "`" + `gitlab_user_gpgkey` + "`" + ` resource allows to manage the lifecycle of a GPG key assigned to the current user or a specific user.
+
+**Upstream API**: [GitLab API docs](https://docs.gitlab.com/ee/api/users.html#get-a-specific-gpg-key)`,
+
+		CreateContext: resourceGitlabUserGPGKeyCreate,
+		ReadContext:   resourceGitlabUserGPGKeyRead,
+		DeleteContext: resourceGitlabUserGPGKeyDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"user_id": {
+				Description: "The ID of the user to add the GPG key to. If this field is omitted, this resource manages a GPG key for the current user. Otherwise, this resource manages a GPG key for the speicifed user, and an admin token is required.",
+				Type:        schema.TypeInt,
+				ForceNew:    true,
+				Optional:    true,
+			},
+			"key": {
+				Description: "The armored GPG public key.",
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Required:    true,
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					strippedOld := strings.TrimSpace(old)
+					strippedNew := strings.TrimSpace(new)
+					return strippedOld == strippedNew
+				},
+			},
+			"key_id": {
+				Description: "The ID of the GPG key.",
+				Type:        schema.TypeInt,
+				Computed:    true,
+			},
+			"created_at": {
+				Description: "The time when this key was created in GitLab.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+		},
+	}
+})
+
+func resourceGitlabUserGPGKeyCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*gitlab.Client)
+
+	options := &gitlab.AddGPGKeyOptions{
+		Key:   gitlab.String(strings.TrimSpace(d.Get("key").(string))),
+	}
+
+	var key *gitlab.GPGKey
+	var err error
+	userID, userIDOk := d.GetOk("user_id")
+	if userIDOk {
+		isAdmin, isAdminErr := isCurrentUserAdmin(ctx, client)
+		if isAdminErr != nil {
```

Comment:
NIT: (let's stick to the `err` idiom :)) 

```suggestion
		isAdmin, err := isCurrentUserAdmin(ctx, client)
		if err != nil {
```